src/app/core/signup/signup.component.ts

```diff
@@ -1,15 +1,39 @@
 import { Component, OnInit } from '@angular/core';
-
+import { AuthService } from '../backend_assets/auth.service';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { User } from '../backend_assets/user.service';
 @Component({
   selector: 'app-signup',
   templateUrl: './signup.component.html',
   styleUrls: ['./signup.component.css']
 })
 export class SignupComponent implements OnInit {
-
-  constructor() { }
-
-  ngOnInit() {
+  signupForm: FormGroup;
+  error: string;
+  user: User;
+  expiration_date: Date;
+  constructor(private mAuth: AuthService, private formBuilder: FormBuilder, private router: Router) {
+    this.signupForm = this.formBuilder.group({ username: ['', Validators.required], password: ['', Validators.required] });
```

Comment:
don't add this in constructor, only dependency injections should be a part of constructor